components/calendar/src/any_calendar.rs

```diff
@@ -608,14 +607,14 @@ impl AnyCalendarKind {
     /// Construct from a BCP-47 string
     ///
     /// Returns None if the calendar is unknown
-    pub fn from_bcp47_string(x: &str) -> Option<Self> {
-        Self::from_bcp47_bytes(x.as_bytes())
+    pub fn try_from_bcp47_string(x: &str) -> Result<Self, DateTimeError> {
```

Comment:
issue: I was actually hoping to go the other way here. Part of the problem here is that `DateTimeError::unknown_kind_from_bytes()` allocates, and if you're immediately throwing away the error it's wasted work. I'd prefer if these all uniformly returned `Option`, and callers did `.ok_or_else(|| DateTimeError::unknown_kind_from_bytes(s))` when necessary.